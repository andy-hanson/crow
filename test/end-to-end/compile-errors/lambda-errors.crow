main void()
	d = () => 1
	3 foo (x string) => ()
	3 bar x => ()

foo void(x nat, f void mut(x nat))
	()
bar void(x nat, f void mut(x nat))
	()
bar void(x nat, f nat)
	()
bar void(x nat, f void mut(x string))
	()
