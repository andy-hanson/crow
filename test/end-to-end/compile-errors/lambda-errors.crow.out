exit code: 1
stderr:
test/end-to-end/compile-errors/lambda-errors.crow 2:9-2:16 Lambda expression needs an expected type.
test/end-to-end/compile-errors/lambda-errors.crow 3:11-3:27 The lambda doesn't match the expected type at this location.
Expected type 'void mut(nat64)'.
test/end-to-end/compile-errors/lambda-errors.crow 4:11-4:18 Multiple lambda types are possible:
	'void mut(nat64)'
	'void mut(string)'
Consider explicitly typing the lambda's parameter.
Uncaught exception: Reached compile error
	at main
